wee_slack.py

```diff
@@ -1386,142 +1415,148 @@ def create_reaction_string(reactions):
         reaction_string = ''
     return reaction_string
 
-# deprecated in favor of redrawing the entire buffer
-#def modify_buffer_line(buffer, user, new_message, time, append):
-#    time = int(float(time))
-#    own_lines = w.hdata_pointer(w.hdata_get('buffer'), buffer, 'own_lines')
-#    if own_lines:
-#        line = w.hdata_pointer(w.hdata_get('lines'), own_lines, 'last_line')
-#        hdata_line = w.hdata_get('line')
-#        hdata_line_data = w.hdata_get('line_data')
-#
-#        while line:
-#            data = w.hdata_pointer(hdata_line, line, 'data')
-#            if data:
-#                date = w.hdata_time(hdata_line_data, data, 'date')
-#                prefix = w.hdata_string(hdata_line_data, data, 'prefix')
-#                if new_message == "":
-#                    new_message = w.hdata_string(hdata_line_data, data, 'message')
-#                if user and (int(date) == int(time) and user == prefix):
-#                    w.prnt("", "found matching time date is {}, time is {} ".format(date, time))
-#                    w.hdata_update(hdata_line_data, data, {"message": "{}{}".format(new_message, append)})
-#                    break
-#                elif not user and (int(date) == int(time)):
-#                    w.hdata_update(hdata_line_data, data, {"message": "{}{}".format(new_message, append)})
-#                else:
-#                    pass
-#            line = w.hdata_move(hdata_line, line, -1)
-#    return w.WEECHAT_RC_OK
-
+def modify_buffer_line(buffer, new_line, time):
+    time = int(float(time))
+    # get a pointer to this buffer's lines
+    own_lines = w.hdata_pointer(w.hdata_get('buffer'), buffer, 'own_lines')
+    if own_lines:
+        #get a pointer to the last line
+        line_pointer = w.hdata_pointer(w.hdata_get('lines'), own_lines, 'last_line')
+        #hold the structure of a line and of line data
+        struct_hdata_line = w.hdata_get('line')
+        struct_hdata_line_data = w.hdata_get('line_data')
+
+        while line_pointer:
+            #get a pointer to the data in line_pointer via layout of struct_hdata_line
+            data = w.hdata_pointer(struct_hdata_line, line_pointer, 'data')
+            if data:
+                date = w.hdata_time(struct_hdata_line_data, data, 'date')
+                prefix = w.hdata_string(struct_hdata_line_data, data, 'prefix')
+
+                if int(date) == int(time):
+                    #w.prnt("", "found matching time date is {}, time is {} ".format(date, time))
+                    w.hdata_update(struct_hdata_line_data, data, {"message": new_line})
+                    break
+                else:
+                    pass
+            #move backwards one line and try again - exit the while if you hit the end
+            line_pointer = w.hdata_move(struct_hdata_line, line_pointer, -1)
+    return w.WEECHAT_RC_OK
 
-def process_message(message_json, cache=True):
+def render_message(message_json, force=False):
```

Comment:
the new render_message function, which takes over for process_message() in many instances.